### PR TITLE
feat: object preferences and unset channel data

### PR DESCRIPTION
**Pull request by cjbell**

### CODE REVIEW

File: lib/knock/objects.rb [hunk 1]
```diff
@@ -137,6 +153,145 @@ def get_messages(collection:, id:, options: {})
 
         execute_request(request: request)
       end
+
+      ##
+      # Preferences
+      ##
+
+      # Returns all preference sets for the user
```
Comment (juanazam) [inline lib/knock/objects.rb:161]: This comment references user instead of object

File: lib/knock/objects.rb [hunk 2]
```diff
@@ -137,6 +153,145 @@ def get_messages(collection:, id:, options: {})
...
+        request = get_request(
+          auth: true,
+          path: endpoint
+        )
+
+        execute_request(request: request)
+      end
+
+      # Gets a single preference set, defaults to the 'default' set
+      # for the user given.
```
Comment (juanazam) [inline lib/knock/objects.rb:178]: user -> object

File: lib/knock/objects.rb [hunk 3]
```diff
@@ -137,6 +153,145 @@ def get_messages(collection:, id:, options: {})
...
+      #
+      # @param [String] collection The collection the object is in
+      # @param [String] id The object id
+      # @param [String] preference_set The preference set ID (defaults to `default`)
+      # @param [String] workflow The workflow to set preferences for
+      # @param [Bool | Hash] setting Either a boolean to indicate if the type is enabled
+      #  or a hash containing channel types and settings
+      #
+      # @return [Hash] The preference set
+      def set_workflow_preferences(collection:, id:, workflow:, setting:, preference_set: DEFAULT_PREFERENCE_SET_ID)
```
Comment (juanazam) [inline lib/knock/objects.rb:260]: `DEFAULT_PREFERENCE_SET_ID` is not defined on this file (it blows up when running)

Comment (cjbell) [reply lib/knock/objects.rb:260]: 🤦 will fix!

File: lib/knock/users.rb [hunk 4]
```diff
@@ -317,6 +317,22 @@ def set_channel_data(id:, channel_id:, channel_data:)
...
       end
 
+      # Unsets user's channel data for the given channel id
+      #
+      # @param [String] id the user ID
+      # @param [String] channel_id target channel ID
+      # @param [Hash] channel_data channel data
+      #
+      # @return [Hash] channel data
+      def unset_channel_data(id:, channel_id:)
```
Comment (juanazam) [inline lib/knock/users.rb:327]: This one is failing for me:

```
irb(main):025:0> Knock::Objects.unset_channel_data(collection: "asset", id: "1", channel_id: "21eb901c-54f6-44ca-a7e9-4f4503d60d36")
/Users/juanazam/.asdf/installs/ruby/3.0.2/lib/ruby/3.0.0/json/common.rb:216:in `initialize': no implicit conversion of nil into String (TypeError)
```

Comment (juanazam) [reply lib/knock/users.rb:327]: I think it fails because the delete request doesn't contain any body on the response

Comment (cjbell) [reply lib/knock/users.rb:327]: 🤔 that means our other delete requests are failing too. Will fix.

Comment (juanazam) [reply lib/knock/users.rb:327]: updating https://github.com/knocklabs/knock-ruby/blob/main/lib/knock/client.rb#L21 to

```
JSON.parse(response.body) if response.body
```

Does the trick

Comment (juanazam) [reply lib/knock/users.rb:327]: @cjbell yep, they are

File: lib/knock/client.rb [hunk 5]
```diff
@@ -18,7 +18,11 @@ def execute_request(request:)
       http_status = response.code.to_i
       handle_error_response(response: response) if http_status >= 400
 
-      JSON.parse(response.body)
+      if response.body and response.body != "" do
+        JSON.parse(response.body) 
+      else
+        nil
```
Comment (juanazam) [inline lib/knock/client.rb:24]: nit: no need for the else here, an `if` statement returns nil if condition doesn't match and there is no else statement

**Review by juanazam — APPROVED** [review]
Looks good, left a small `nit` comment

**Pull request merged**